### DISCUSSION TITLE
Progress is not shown when using jsPsych.pluginAPI.preloadAudioFiles

### DISCUSSION THREAD

**Discussion by zhangchi199133** [jspsych/jsPsych#959] [Q&A] [answered]
Hi all,
I tried to make a display of the progress of audio file preloading by mimicking the codes in https://www.jspsych.org/core_library/jspsych-pluginAPI/#jspsychpluginapipreloadaudiofiles. The related codes are as follows:

```js
`      var preload_audio_trial = {
            timeline_variables: trial_stimuli,
            timeline: [
              {
            type: "call-function",
            async: true,
            func: function (done) {
                jsPsych.pluginAPI.preloadAudioFiles([jsPsych.timelineVariable('audio', true)],
                function () {done({ preload: "success" })},
                function(nLoaded) {updateLoadedCount(nLoaded);},
              );
            },
            on_finish: function updateLoadedCount(nLoaded){
              var percentcomplete = nLoaded / audio.length * 100;
              console.log('Loaded '+percentcomplete+'% of audio files');
            }
          },
}`
```
However, when I ran them, I got errors as such:

Uncaught ReferenceError: updateLoadedCount is not defined
    at experiment.js:587
    at AudioContext.<anonymous> (jspsych.js:2308)

I don't know where my codes went wrong. Would you please help me with this? Thank you so much!

Best regards,

Chi

**Comment by jodeleeuw** [comment, marked as answer]
Hi Chi,

To draw the a progress bar on the screen you'd need to add some code that creates a DOM element to display the progress bar. In `jspsych.js` that's done here:

https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/jspsych.js#L1000-L1022

Your current code creates an error because you are trying to call the `updateLoadedCount()` function, but this function doesn't exist. You've copied that function into the `on_finish` event for the timeline, but this won't create the function in a way that is accessible to the function you've defined in `func`.

**Reply by zhangchi199133** [reply]
Thanks a lot!
